certeasy-backend-app/src/main/java/org/certeasy/backend/issuer/CertIssuer.java

```diff
@@ -83,6 +83,12 @@ public Optional<Certificate> getCertificate() {
         return Optional.ofNullable(certificate);
     }
 
+    public void deleteIssuedCert(StoredCert storedCert){
+        if(storedCert.getCertificate().getSerial().equals(serial))
+            throw new ReadOnlyCertificateException(storedCert.getCertificate().
+                    getSerial());
```

Comment:
am having a loop trying to understand this condition.
does this mean that the certificate cant be deleted if it is the issuer's own certificate?

Reply:
Exactly, You cannot delete the issuer's certificate. You can only delete the issuer as a whole.

Reply:
Awesome